src/Symfony/Component/Cache/Adapter/PdoAdapter.php

```diff
@@ -108,9 +108,20 @@ public function createTable()
         $conn = $this->getConnection();
 
         if ($conn instanceof Connection) {
+            $types = array(
+                'mysql'  => 'binary',
+                'sqlite' => 'text',
+                'pgsql'  => 'string',
+                'oci'    => 'string',
+                'sqlsrv' => 'string',
+            );
+            if (!isset($types[$this->driver])) {
+                throw new \DomainException(sprintf('Creating the cache table is currently not implemented for PDO driver "%s".', $this->driver));
+            }
+            
             $schema = new Schema();
             $table = $schema->createTable($this->table);
-            $table->addColumn($this->idCol, 'blob', array('length' => 255));
+            $table->addColumn($this->idCol, $types[$this->driver], array('length' => 255));
```

Comment:
I must say that I don't really know why sqlite uses text, but as you wrote it, I hope you do ;). In case this should be varchar too, then sqlite should also use 'string'. Then I would say that something like `'mysql' === $this->driver ? 'binary' : 'string'` would be applicable.